base/sparse/abstractsparse.jl

```diff
@@ -19,5 +19,21 @@ issparse(S::LowerTriangular{<:Any,<:AbstractSparseMatrix}) = true
 issparse(S::LinAlg.UnitLowerTriangular{<:Any,<:AbstractSparseMatrix}) = true
 issparse(S::UpperTriangular{<:Any,<:AbstractSparseMatrix}) = true
 issparse(S::LinAlg.UnitUpperTriangular{<:Any,<:AbstractSparseMatrix}) = true
+issparse(S::Base.ReinterpretArray) = issparse(S.parent)
 
 indtype(S::AbstractSparseArray{<:Any,Ti}) where {Ti} = Ti
+
+nonzeros(A::Base.ReinterpretArray{T}) where {T} = reinterpret(T, nonzeros(A.parent))
+function nonzeroinds(A::Base.ReinterpretArray{T,N,S} where {N}) where {T,S}
+    if sizeof(T) == sizeof(S)
+        return nonzeroinds(A.parent)
+    elseif sizeof(T) > sizeof(S)
+        unique(map(nonzeroinds(A.parent)) do ind
+            div(ind, div(sizeof(T), sizeof(S)))
+        end)
+    else
+        map(nonzeroinds(A.parent)) do ind
+            ind * div(sizeof(S), sizeof(T))
+        end
+    end
+end
```

Comment:
This method assumes that `reinterpret(zero(T)) === zero(S)`. I don't think we should define it.

Reply:
So did the old `reinterpret` kind off, which is why I didn't change it. Happy to get rid of this though.

Reply:
The old definition returned a sparse reinterpreted array (`SparseArray{ReinterpretedArray{parent}}`), the new one returns a reinterpreted sparse array (`Reinterpreted{A}`). The main point where they differ is in the handling of this method, since the old one would return a strong zero (`zero(S)`), where the new one would return a casted zero (`reinterpret(S, zero(T))`)

Reply:
That's true, but since nonzeroinds returns structural non-zeros, that still seems fine to (if odd if `reinterpret(S, zero(T)) != zero(S)`. Would be good to have some input from people who care about sparse matrices.